src/raftstore/store/snap.rs

```diff
@@ -2318,8 +2318,29 @@ mod test {
         let snap_mgr = SnapManagerBuilder::default()
             .max_total_size(10240)
             .build(snapfiles_path.path().to_str().unwrap(), None);
-
         let snapshot = DbSnapshot::new(kv);
+
+        // Add an oldest snapshot for receiving.
+        let recv_key = SnapKey::new(100, 100, 100);
+        let recv_data = {
+            let mut s = snap_mgr
+                .get_snapshot_for_building(&recv_key, &snapshot)
+                .unwrap();
+            let mut snap_data = RaftSnapshotData::new();
+            let mut stat = SnapshotStatistics::new();
+            s.build(
+                &snapshot,
+                &get_test_region(100, 1, 1),
+                &mut snap_data,
+                &mut stat,
+                Box::new(snap_mgr.clone()),
+            ).unwrap();
+            snap_data.write_to_bytes().unwrap()
+        };
+        snap_mgr
```

Comment:
No snapshot is saved since it's dropped immediately.

Reply:
I think `SnapManager` still holds the record of the snapshot, even if no data are really stored on disk. You can see the CI is passed.